k8-operator/controllers/infisicalsecret_helper.go

```diff
@@ -28,19 +58,102 @@ const OPERATOR_SETTINGS_CONFIGMAP_NAME = "infisical-config"
 const OPERATOR_SETTINGS_CONFIGMAP_NAMESPACE = "infisical-operator-system"
 const INFISICAL_DOMAIN = "https://app.infisical.com/api"
 
-type AuthStrategyType string
+var infisicalClient = infisicalSdk.NewInfisicalClient(infisicalSdk.Config{})
+var authDetails AuthenticationDetails
 
-var AuthStrategy = struct {
-	SERVICE_TOKEN              AuthStrategyType
-	SERVICE_ACCOUNT            AuthStrategyType
-	UNIVERSAL_MACHINE_IDENTITY AuthStrategyType
-}{
-	SERVICE_TOKEN:              "SERVICE_TOKEN",
-	SERVICE_ACCOUNT:            "SERVICE_ACCOUNT",
-	UNIVERSAL_MACHINE_IDENTITY: "UNIVERSAL_MACHINE_IDENTITY",
-}
+func (r *InfisicalSecretReconciler) HandleAuthentication(ctx context.Context, infisicalSecret v1alpha1.InfisicalSecret) (AuthenticationDetails, error) {
```

Comment:
To make this function bit easier to read, you should make each auth method it's own standalone method of `*InfisicalSecretReconciler` and then make a map of auth type => auth handler function. Something like this so we don't have a ton of if conditions: and things are a bit more modular:

```
func (r *InfisicalSecretReconciler) HandleAuthentication(ctx context.Context, infisicalSecret v1alpha1.InfisicalSecret) (AuthenticationDetails, error) {
	authStrategies := map[AuthStrategyType]func(ctx context.Context, infisicalSecret v1alpha1.InfisicalSecret) (AuthenticationDetails, error){
		AuthStrategy.UNIVERSAL_MACHINE_IDENTITY:    r.handleUniversalAuth,
		AuthStrategy.KUBERNETES_MACHINE_IDENTITY:   r.handleKubernetesAuth,
        ...
	}

	for strategy, handler := range authStrategies {
		authDetails, err := handler(ctx, infisicalSecret)
		if err == nil {
			return authDetails, nil
		}
		if err != nil && !errors.Is(err, AuthNotApplicableError) {
			return AuthenticationDetails{}, fmt.Errorf("authentication failed for strategy.......")
		}
	}

	return AuthenticationDetails{}, fmt.Errorf("no valid authentication strategy found")
}

func (r *InfisicalSecretReconciler) handleServiceTokenAuth(ctx context.Context, infisicalSecret v1alpha1.InfisicalSecret) (AuthenticationDetails, error) {
	infisicalToken, err := r.GetInfisicalTokenFromKubeSecret(ctx, infisicalSecret)
	if err != nil || infisicalToken == "" {
		return AuthenticationDetails{}, AuthNotApplicableError
	}
	return AuthenticationDetails{authStrategy: AuthStrategy.SERVICE_TOKEN}, nil
}
....
```